src/stores/my-account.ts

```diff
@@ -6,19 +6,23 @@ import {
   Signer,
 } from '@/utils/account'
 import { LocalStorage } from '@/utils/storage'
+import { initializeWeb3Auth, SubstrateRPC } from '@/utils/substrate-rpc'
+import { Web3Auth } from '@web3auth/modal'
 import { useAnalytics } from './analytics'
 import { create } from './utils'
 
 type State = {
-  isInitialized?: boolean
-  isInitializedAddress?: boolean
+  web3Auth: Web3Auth | null
```

Comment:
for this web3Auth in zustand, I still think its better to have this not in zustand,
I understand what you meant, so you need web3Auth in login, logout, etc, so you want to have only one instance.

so to solve this, you have to create singleton instance (like in here, you actually partially implement it), but it doesn't need to be in zustand store, it can just be global variable.

```
const web3Auth: Web3Auth | null
function getWeb3Auth () {
  if (web3Auth) return web3Auth
  web3Auth = initializeWeb3Auth()
  return web3Auth
}
```

Reply:
I'm already doing this. The initializeWeb3Auth function already handles the creation and I moved the Web3Auth instance outside the state too